src/keri/core/kraming.py

```diff
@@ -0,0 +1,231 @@
+                reason (str): A description of why the message was accepted or rejected
+        """
+        if not serder.verify():
+            return False, "Invalid message structure"
+
+        sad = serder.sad
+
+        sourceAid = sad.get("i", None)
+        messageType = serder.ilk or None
+        timestamp = sad.get("dt", None)
+
+        if not all([sourceAid, messageType, timestamp]):
+            return False, "Missing required message fields"
+
+        windowSize, driftSkew = self.getWindowParameters(sourceAid, messageType)
+
+        # Convert both timestamps to seconds since epoch for comparison
+        currentTime = helping.fromIso8601(helping.nowIso8601()).timestamp()
+
+        messageTime = helping.fromIso8601(timestamp).timestamp()
+
+        if (messageTime < currentTime - driftSkew - windowSize or
+                messageTime > currentTime + driftSkew):
+            raise kering.ValidationError(f"Message is out of time window {serder.pretty()}")
+
+        cacheKey = self._constructCacheKey(serder)
+
+        cachedTimestamp = self._getCachedTimestamp(cacheKey)
+
+        if cachedTimestamp is None:
+            self._storeTimestamp(cacheKey, messageTime)
+            # Message accepted, new entry
+            return True
+
+        if messageTime > cachedTimestamp:
+            self._storeTimestamp(cacheKey, messageTime)
+            # Message accepted, updated cached entry
+            return True
+
+        if messageTime == cachedTimestamp:
+            raise kering.ValidationError(f"Message replay detected {serder.pretty()}")
+
+        raise kering.ValidationError(f"Message is out of order {serder.pretty()}")
+
+    def pruneCache(self):
+        """Prune stale entries from the Replay Cache Table.
+
+        Returns:
+            int: The number of pruned entries
+        """
+        prunedCount = 0
+        currentTime = helping.fromIso8601(helping.nowIso8601()).timestamp()
+
+        for key, timestampStr in self.db.time.getItemIter():
+            try:
+
+                timestamp = float(timestampStr)
+
+                windowSize, driftSkew = self.getWindowParameters(key)
+
+                if timestamp < currentTime - driftSkew - windowSize:
+                    self.db.time.rem(key)
+                    prunedCount += 1
+
+            except Exception:
+                continue
+
+        return prunedCount
+
+    def processKrms(self):
+        for said, serder in self.db.krms.getFullItemIter():
+            try:
+                # assumes serder is a SerderKERI, more processing may be needed
+                if self.checkMessageTimeliness(serder):
+                    # TODO: Implement escrowing functionality
+                    self.db.krms.rem(said)
+                    logger.info(f"Message accepted: {serder.pretty()}")
+            except kering.ValidationError as e:
+                logger.error(f"Invalid message: {e}")
+            self.pruneCache()
```

Comment:
Should this `pruneCache()` call occur no matter what is in the `.krms` DB? As in, should it be shifted to the left by one indentation level?

Another way of saying it is should prune cache only occur when there are items in the `.krms` DB or should it always occur on each recur of the KramDoer?